File: app/src/main/java/cat/xojan/random1/viewmodel/ProgramViewModel.java
package cat.xojan.random1.viewmodel;

import android.databinding.BaseObservable;
import android.view.View;

import cat.xojan.random1.domain.entities.Program;
import cat.xojan.random1.domain.interactor.ProgramDataInteractor;
import cat.xojan.random1.ui.activity.BaseActivity;
import cat.xojan.random1.ui.fragment.HourByHourListFragment;
import cat.xojan.random1.ui.fragment.SectionFragment;

public class ProgramViewModel extends BaseObservable {

    private final BaseActivity mActivity;
    private final Program mProgram;
    private final ProgramDataInteractor mProgramDataInteractor;

    public ProgramViewModel(BaseActivity activity, Program program,
                            ProgramDataInteractor programDataInteractor) {
        mActivity = activity;
        mProgram = program;
        mProgramDataInteractor = programDataInteractor;
    }

    public View.OnClickListener onClickProgram() {
        return v -> {
            if (mProgramDataInteractor.isSectionSelected() && mProgram.getSections().size() > 1) {
                SectionFragment sectionListFragment = SectionFragment.newInstance(mProgram);
                mActivity.addFragment(sectionListFragment, SectionFragment.TAG, true);
            } else {
                HourByHourListFragment hourByHourListFragment =
                        HourByHourListFragment.newInstance(mProgram);
                mActivity.addFragment(hourByHourListFragment, HourByHourListFragment.TAG, true);
            }
        };
    }

    public String getImageUrl() {
        return mProgram.getImageUrl();
    }

    public String getTitle() {
        return mProgram.getTitle();
    }
}

File: app/src/main/java/cat/xojan/random1/viewmodel/ProgramViewModel.kt
package cat.xojan.random1.viewmodel

import android.databinding.BaseObservable
import android.view.View
import cat.xojan.random1.domain.entities.Program
import cat.xojan.random1.domain.interactor.ProgramDataInteractor
import cat.xojan.random1.ui.activity.BaseActivity
import cat.xojan.random1.ui.activity.BrowseActivity


class ProgramViewModel(
        private val activity: BaseActivity,
        private val program: Program,
        private val programDataInteractor: ProgramDataInteractor
) : BaseObservable() {

    val imageUrl: String?
        get() = program.imageUrl

    val title: String
        get() = program.title

    fun onClickProgram(): View.OnClickListener {
        return View.OnClickListener {
            val isSection = programDataInteractor.isSectionSelected && program.sections.size > 1
            val intent = BrowseActivity.newIntent(activity, program, isSection)
            activity.startActivity(intent)
        }
    }
}